ruby/phase-1.md

```diff
@@ -0,0 +1,575 @@
+# Ruby SDK - Phase 1
+
+Ruby is a popular dynamic programming language that many companies are leaning on to beat their
+competitors to the market. With the focus on the speed of execution it is very common for these
+companies to rely on managed solutions. This presents an opportunity for Temporal to become Ruby's
+de facto standard for workflow orchestration.
+
+
+## Timeline
+
+The Ruby SDK will be implemented in four loosely defined phases:
+
+- Phase 1 — initial setup, activity APIs, client and worker, payload converters
+- Phase 2 — per-class execution attributes, logging, metrics, error handling, middleware
+- Phase 3 — workflow APIs, local activities, async activity completion, etc
+- Phase 4 — Sandbox, replayer, support for testing
+
+
+## Overview
+
+The Ruby SDK will be implemented on top of the [SDK Core](https://github.com/temporalio/sdk-core).
+The exact bridging mechanism will be decided later as there are different alternatives.
+
+### Goals
+
+- Quick path to a full working setup
+- A path of minimal divergence from other SDKs (does allow for other approaches, but it's not the
+  goal)
+- Minimise external dependencies (gems)
+- Local and end-to-end testability of workflows
+- Type safety where possible (optional and exact approach to be chosen later)
+- Compatibility with existing tools (error handling, logging, metrics, middleware, etc)
+
+### Non-goals
+
+- Convention over configuration (in favour of cross-SDK interface consistency, this goes against
```

Comment:
?? I'm confused on this point. We can still provide a near clone of the Go/Java/Typescript APIs for the low level SDK methods while also wrapping them up in some higher order classes that developers will want to use. Is this in the right category?